charts/astronomer/templates/_helpers.yaml

```diff
@@ -41,6 +41,22 @@
 {{- end }}
 {{- end }}
 
+{{ define "astronomer.platform_registry" -}}
+{{- if .Values.global.privateRegistry.enabled -}}
+{{ .Values.global.privateRegistry.repository }}
+{{- else -}}
+quay.io/astronomer
+{{- end }}
+{{- end }}
+
+{{ define "authSidecar.image" -}}
+{{- if .Values.global.authSidecar.repository -}}
+{{ .Values.global.authSidecar.repository }}
+{{- else -}}
+{{ template "astronomer.platform_registry" . }}/ap-nginx-unprivileged
```

Comment:
This does not account for variety of the image name and does not include an image tag.

Reply:
Starting to dust off this ticket including giving it a proper description and existing the related upstream issue. Regarding this specific template - the optional .Values.global.authSidecar.repository (when set) must include the full path to the image and so users can over-ride the image name there. I wouldn't anticipate we'd want the portion after the default location to change - part of the purpose of this is to say 'look mirror everything here and mirror it with the existing names if you don't want to have to configure anything.'

Reply:
@danielhoherd Added nine new tests, please review at your convenience.

Reply:
@rob-1126 all of the other places we do this kind of thing are here: https://github.com/astronomer/astronomer/blob/master/charts/astronomer/templates/_helpers.yaml#L12-L58

is there some reason we can't re-use that simpler snip of code that is working elsewhere?